src/explorer/Common/IndexedGroupingSetsResultMulti.cs

```diff
@@ -0,0 +1,41 @@
+namespace Explorer.Common
+{
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Text.Json;
+
+    using Diffix;
+    using Explorer.JsonExtensions;
+
+    public class IndexedGroupingSetsResultMulti<TIndex> : CountableRow
+    {
+        internal IndexedGroupingSetsResultMulti(ref Utf8JsonReader reader, ImmutableArray<TIndex> groupingLabels)
+        {
+            AllGroupingLabels = groupingLabels;
+            (GroupingId, Values) = reader.ParseMultiGroupingSet(groupingLabels.Length);
+            Count = reader.ParseCount();
+            CountNoise = reader.ParseCountNoise();
+        }
+
+        public int GroupingId { get; }
+
+        public ImmutableArray<DValue<JsonElement>> Values { get; }
+
+        public long Count { get; }
+
+        public double? CountNoise { get; }
+
+        public IEnumerable<TIndex> GroupingLabels
+            => AllGroupingLabels.Where((_, index) => GroupingIndices.Contains(index));
+
+        public IEnumerable<int> GroupingIndices =>
+            GroupingIdConverter.GetConverter(AllGroupingLabels.Length).IndicesFromGroupingId(GroupingId);
+
+        public bool IsNull => Values.All(v => v.IsNull);
+
+        public bool IsSuppressed => Values.Any(v => v.IsSuppressed);
```

Comment:
It is not clear why for `IsNull` you use `All` and here you use `Any`.

Reply:
I wasn't 100% sure about this but I think it makes sense intuitively:
- a row can be considered null if all of the columns are null (== if any column has a value, the row has a value). 
- suppressed if any of the columns are suppressed (== if all columns are clear, the row is clear). 

Also both members are part of the `CountableRow` interface so they need to be implemented somehow...

Reply:
For me both statements also make sense if I replace *any* with *all* and *all* with *any*:
- a row can be considered null if *all* of the columns are null (== if any column has a value, the row has a value).
- a row can be considered suppressed if *any* of the columns are suppressed (== if all columns are clear, the row is clear).
VS
- a row can be considered null if *any* of the columns is null
- a row can be considered suppressed if *all* of the columns are suppressed

Reply:
I'll try to explain a little better:

Null is like nothing. 
If all columns have nothing, then the row has nothing. 
But if any column has *something* (not null) then the row has *something*. 

Suppressed values are uncertain. 
If any column is uncertain than that makes the row uncertain. 
Only if all columns are *certain* (not suppressed) can the row be considered certain.

Does it make more sense like this?

Reply:
Yes, this is helping me to understand the idea. Maybe it will be helpful to also add these explanations as comments in the code.